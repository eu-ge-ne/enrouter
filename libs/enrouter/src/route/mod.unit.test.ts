import { describe, test, expect } from "vitest";

import { buildRoutes } from "./mod.js";

import type { BuildRoutesParams } from "./mod.js";

describe("buildRoutes", () => {
  test("from 0 modules", () => {
    const params: BuildRoutesParams = {
      entryId: "",
      modules: {},
      assets: {},
    };

    expect(buildRoutes(params)).toMatchSnapshot();
  });

  test("from 1 module", () => {
    const params: BuildRoutesParams = {
      entryId: "src/main.tsx",
      modules: {
        "src/_layout.tsx": {
          path: "_layout.tsx",
          load: async () => undefined,
        },
      },
      assets: {
        "src/main.tsx": { modules: ["src/main.tsx"], styles: [] },
        "src/_layout.tsx": { modules: ["src/_layout.tsx"], styles: [] },
      },
    };

    expect(buildRoutes(params)).toMatchSnapshot();
  });

  test("from 2 parent-child modules", () => {
    const params: BuildRoutesParams = {
      entryId: "src/main.tsx",
      modules: {
        "src/_layout.tsx": {
          path: "_layout.tsx",
          load: async () => undefined,
        },
        "src/abc/_layout.tsx": {
          path: "abc/_layout.tsx",
          load: async () => undefined,
        },
      },
      assets: {
        "src/main.tsx": { modules: ["src/main.tsx"], styles: [] },
        "src/_layout.tsx": { modules: ["src/_layout.tsx"], styles: [] },
        "src/abc/_layout.tsx": { modules: ["src/abc/_layout.tsx"], styles: [] },
      },
    };

    expect(buildRoutes(params)).toMatchSnapshot();
  });
});
