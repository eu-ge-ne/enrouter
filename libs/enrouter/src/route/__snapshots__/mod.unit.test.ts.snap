// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildRoutesWithViteManifest > from 0 modules 1`] = `undefined`;

exports[`buildRoutesWithViteManifest > from 1 module 1`] = `
{
  "link": [
    [
      "assets/main.css",
    ],
    [
      "assets/main.js",
      "assets/_layout.js",
      "assets/log.js",
    ],
  ],
  "mod": [
    "src/_layout.tsx",
  ],
  "path": "/",
}
`;

exports[`buildRoutesWithViteManifest > from 2 parent-child modules 1`] = `
{
  "link": [
    [
      "assets/main.css",
    ],
    [
      "assets/main.js",
      "assets/_layout.js",
      "assets/log.js",
    ],
  ],
  "mod": [
    "src/_layout.tsx",
  ],
  "path": "/",
  "tree": [
    {
      "link": [
        [],
        [
          "assets/abc-_layout.js",
          "assets/log.js",
        ],
      ],
      "mod": [
        "src/abc/_layout.tsx",
      ],
      "path": "/abc",
    },
  ],
}
`;
